packages/main/src/requests/request-helpers.ts

```diff
@@ -104,6 +106,31 @@ function assignRoleToPartsAndValidateSendMessageRequest(
   return functionContent;
 }
 
+export function formatCountTokensInput(
+  params: CountTokensRequest | string | Array<string | Part>,
+  model: string,
+): CountTokensRequestInternal {
+  let formattedRequest: CountTokensRequestInternal = {};
+  const containsGenerateContentRequest =
+    (params as CountTokensRequest).generateContentRequest !== undefined;
+  if ((params as CountTokensRequest).contents) {
+    if (containsGenerateContentRequest) {
+      throw new GoogleGenerativeAIError(
```

Comment:
This is a great place to use the brand new `GoogleGenerativeAIRequestInputError` we are using for bad input. https://github.com/google-gemini/generative-ai-js/blob/main/packages/main/src/errors.ts#L66 Although I am not 100% sure we throw in this situation. Do you want to double check in Gemini chat? What does the backend do? Does it error? Does it ignore one? Which one?

Reply:
Done. And team consensus is to throw an error here (I checked).